scipy/stats/_axis_nan_policy.py

```diff
@@ -31,30 +31,19 @@ def _broadcast_array_shapes(arrays, axis=None):
     return _broadcast_shapes(shapes, axis)
 
 
-def _broadcast_shapes(shapes, axis=None):
+def _process_axis(axis, shapes):
     """
-    Broadcast shapes, ignoring incompatibility of specified axes
+    Validate, remove negative elements, and sort axis tuple
     """
-    if not shapes:
-        return shapes
-
-    # input validation
-    if axis is not None:
+    if (axis is not None) and shapes:
         axis = np.atleast_1d(axis)
         axis_int = axis.astype(int)
         if not np.array_equal(axis_int, axis):
             raise ValueError('`axis` must be an integer, a '
```

Comment:
You mentioned `np.AxisError`. This would be a fine time to switch it.

Reply:
Will change it here.

Reply:
Done